pallets/pallet-bonded-coins/src/tests/transactions/create_pool.rs

```diff
@@ -1 +1,251 @@
+use frame_support::{assert_err, assert_ok};
+use frame_system::{pallet_prelude::OriginFor, RawOrigin};
+use pallet_assets::{Error as AssetsPalletErrors, Event as AssetsPalletEvents};
+use sp_runtime::{ArithmeticError, BoundedVec};
+use sp_std::ops::Sub;
 
+use crate::{
+	mock::runtime::*, mock::*, types::PoolStatus, Event as BondingPalletEvents, NextAssetId, Pools, TokenMetaOf,
+};
+
+// create_pool tests
+
+#[test]
+fn single_currency() {
+	let initial_balance = 100_000_000_000_000_000u128;
+	ExtBuilder::default()
+		.with_native_balances(vec![(ACCOUNT_00, initial_balance)])
+		.with_collaterals(vec![DEFAULT_COLLATERAL_CURRENCY_ID])
+		.build()
+		.execute_with(|| {
+			assert_eq!(NextAssetId::<Test>::get(), 0);
```

Comment:
We usually don't run assertions on state that is created in the mock. If you are to run these assertions, they should be put inside the mock `build` function, so that you are sure calling `build()` gives you the state you expect.

Reply:
makes sense, I added this to make sure we detect relevant changes to the mock state though - do we not do this ?

Reply:
Those changes should be detected inside the mock itself then, not here. Moving those assertions inside the `build` function, as we do here for instance: https://github.com/KILTprotocol/kilt-node/blob/4199b6fed27e37a1e8f774f0772e7667c82bf80d/pallets/pallet-asset-switch/src/mock.rs#L261

Reply:
I'll just remove them for now (63b1b43689fe3f7c95b21837e1975d7d8e07616e). We're not setting the `NextAssetId` at all in the build method, which is probably an issue...